homeassistant/components/mazda/climate.py

```diff
@@ -0,0 +1,187 @@
+        """Initialize Mazda climate entity."""
+        super().__init__(client, coordinator, index)
+
+        self.region = region
+        self._attr_unique_id = self.vin
+
+        if self.data["hvacSetting"]["temperatureUnit"] == "F":
+            self._attr_precision = PRECISION_WHOLE
+            self._attr_temperature_unit = UnitOfTemperature.FAHRENHEIT
+            self._attr_min_temp = 61.0
+            self._attr_max_temp = 83.0
+        else:
+            self._attr_precision = PRECISION_HALVES
+            self._attr_temperature_unit = UnitOfTemperature.CELSIUS
+            if region == "MJO":
+                self._attr_min_temp = 18.5
+                self._attr_max_temp = 31.5
+            else:
+                self._attr_min_temp = 15.5
+                self._attr_max_temp = 28.5
+
+        self._update_state_attributes()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Update attributes when the coordinator data updates."""
+        self._update_state_attributes()
+
+        super()._handle_coordinator_update()
+
+    def _update_state_attributes(self) -> None:
+        # Update the HVAC mode
+        hvac_on = self.client.get_assumed_hvac_mode(self.vehicle_id)
+        self._attr_hvac_mode = HVACMode.HEAT_COOL if hvac_on else HVACMode.OFF
+
+        # Update the target temperature
+        hvac_setting = self.client.get_assumed_hvac_setting(self.vehicle_id)
+        self._attr_target_temperature = hvac_setting.get("temperature")
+
+        # Update the current temperature
+        current_temperature_celsius = self.data["evStatus"]["hvacInfo"][
+            "interiorTemperatureCelsius"
+        ]
+        if self.data["hvacSetting"]["temperatureUnit"] == "F":
+            self._attr_current_temperature = convert_temperature(
+                current_temperature_celsius,
+                UnitOfTemperature.CELSIUS,
+                UnitOfTemperature.FAHRENHEIT,
+            )
+        else:
+            self._attr_current_temperature = current_temperature_celsius
+
+        # Update the preset mode based on the state of the front and rear defrosters
+        front_defroster = hvac_setting.get("frontDefroster")
+        rear_defroster = hvac_setting.get("rearDefroster")
+        if front_defroster and rear_defroster:
+            self._attr_preset_mode = PRESET_DEFROSTER_FRONT_AND_REAR
+        elif front_defroster:
+            self._attr_preset_mode = PRESET_DEFROSTER_FRONT
+        elif rear_defroster:
+            self._attr_preset_mode = PRESET_DEFROSTER_REAR
+        else:
+            self._attr_preset_mode = PRESET_DEFROSTER_OFF
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set a new HVAC mode."""
+        if hvac_mode == HVACMode.HEAT_COOL:
+            await self.client.turn_on_hvac(self.vehicle_id)
+        elif hvac_mode == HVACMode.OFF:
+            await self.client.turn_off_hvac(self.vehicle_id)
+
+        self._handle_coordinator_update()
```

Comment:
It's a bit weird to call this method when there's no coordinator update. I'd create a different helper that we can call that calls `self._update_state_attributes` and `self.async_write_ha_state`.